worker/utils/utils.js

```diff
@@ -49,6 +51,27 @@ module.exports = {
     })
   },
 
+  async encryptJob(string1, string2){
+    const secret = this.retrievePassword() + string1 + string2
+    const salt = this.generateSalt();
+    const digest = crypto.scryptSync(secret, salt, 64);
+    return digest.toString('hex');
+  },
+
+  async decryptJob(digest, string1, string2){
+
```

Comment:
Since we aren't decrypting, you could call this "validateJob", and just call encryptJob within. Something like 
crypto.timingSafeEqual(digest, encryptJob(string1, string2);

More about timing attack protection -- there are other vulnerabilities to this approach that I'd be more worried about but Andrew reminded me it's good practice. https://nodejs.org/api/crypto.html#crypto_crypto_timingsafeequal_a_b